src/realm/sync/network/default_socket.cpp

```diff
@@ -471,18 +471,19 @@ DefaultSocketProvider::DefaultSocketProvider(const std::shared_ptr<util::Logger>
     REALM_ASSERT(m_logger_ptr);                     // Make sure the logger is valid
     util::seed_prng_nondeterministically(m_random); // Throws
     start_keep_running_timer();                     // TODO: Update service so this timer is not needed
-    start();
+    if (auto_start) {
+        start();
+    }
 }
 
 DefaultSocketProvider::~DefaultSocketProvider()
 {
-    // Wait for the thread to stop
-    stop(true);
+    m_logger_ptr->trace("Default event loop teardown");
     if (m_keep_running_timer)
         m_keep_running_timer->cancel();
-    if (m_thread.joinable()) {
```

Comment:
I think stop() should be the thing that joins the worker thread

Reply:
But only if `wait_for_stop` is true.  Sometimes `client.stop()`, which calls `socket_provider.stop()` (for now) is called on the event loop thread.

Reply:
Right, but we're calling `stop(true)` here. I guess my comment there means "we're calling stop(true), which calls join if wait_for_stop is true, which it is, so we don't need to join here."

Reply:
In other words, we should either just detach the thread so that joining it doesn't matter, or we should join it when stop is called with `wait_for_stop==true`.